### ISSUE TITLE
Test AS and TS version of serializer

### ISSUE THREAD

**Issue by peterjah** [massalabs/as#82]
**Context**
The argument class as evolved and is not compatible anymore with the Args class in Typescript from massa-web3.
Another issue will be created in web3 repo to fix this.
This issue is about detection of such differences between the assemblyscript and the typescript version of Args.

**Proposed solution**
Import massa-web3 lib as dev dependency and use it to generate a file containing both serialized and not serialized values for each types. For convenient use, this file can be commited, alongside its generation script. This file will be used as input for testing serializer in assemblyscript.

This solution is not perfect but it will allow us to detect a breaking change made in the AS side.
The drawback is to detect a breaking change in the TS side, the massa-web3 lib must be updated and the testfile, regenerated

**Comment by peterjah** [comment]
part of this work is already done! contact me for more details :)

**Comment by gregLibert** [comment]
@longgamma-massa should we spend more time on that or strat testing the contract abi that will obsolete all this ?

**Comment by longgamma-massa** [comment]
> @longgamma-massa should we spend more time on that or strat testing the contract abi that will obsolete all this ?

Option 2: Start testing pre-compile step + contract ABI since it unblocks the ability to interact with deployed (others) contracts, and removes serialization on user-side.

**Comment by Ben-Rey** [comment]
@gregLibert can we close this issue ?
